alm/src/services/SnapshotProvider.ts

```diff
@@ -0,0 +1,75 @@
+const initialValue = {
+  chainId: 0,
+  RequiredBlockConfirmationChanged: [],
+  validatorAddress: '',
+  RequiredSignaturesChanged: [],
+  ValidatorAdded: [],
+  ValidatorRemoved: [],
+  snapshotBlockNumber: 0
+}
+
+export interface SnapshotEvent {
+  blockNumber: number
+  returnValues: any
+}
+
+export interface SnapshotValidatorEvent {
+  blockNumber: number
+  returnValues: any
+  event: string
+}
+
+export interface Snapshot {
+  chainId: number
+  RequiredBlockConfirmationChanged: SnapshotEvent[]
+  validatorAddress: string
+  RequiredSignaturesChanged: SnapshotEvent[]
+  ValidatorAdded: SnapshotValidatorEvent[]
+  ValidatorRemoved: SnapshotValidatorEvent[]
```

Comment:
Could you estimate the growth of the snapshot if we include all transactions from the bridge validators to it? What information received from the chains by the ALM could be included in the snapshot as well?

Reply:
This is how the snapshot looks like for the sokol - kovan amb bridge for the home side
```
{
  "snapshotBlockNumber": 15515659,
  "chainId": 77,
  "RequiredBlockConfirmationChanged": [
    {
      "blockNumber": 9849622,
      "returnValues": {
        "requiredBlockConfirmations": "1"
      }
    }
  ],
  "validatorAddress": "0x0c7A04cc9B1fF1184c5bc7253869727f29593465",
  "RequiredSignaturesChanged": [
    {
      "blockNumber": 9849617,
      "returnValues": {
        "requiredSignatures": "1"
      }
    },
    {
      "blockNumber": 15503549,
      "returnValues": {
        "requiredSignatures": "2"
      }
    }
  ],
  "ValidatorAdded": [
    {
      "blockNumber": 9849617,
      "returnValues": {
        "validator": "0x5d1486951B03A4a8AAdbd9956709041B48003Ea0"
      },
      "event": "ValidatorAdded"
    },
    {
      "blockNumber": 15501966,
      "returnValues": {
        "validator": "0xC5f399f0110E62B9F98fA7e48b66E49216daA598"
      },
      "event": "ValidatorAdded"
    },
    {
      "blockNumber": 15501970,
      "returnValues": {
        "validator": "0x37d95122Cc03CF76a52DEd96f03E51F10c655635"
      },
      "event": "ValidatorAdded"
    }
  ],
  "ValidatorRemoved": []
}
```
It includes all the information related to the bridge contracts that the ALM needs. The only information not stored is about the validator operations transactions, tx receipts, and blocks data. I can try to estimate the growth if we include all the transactions from the validator to sign and execute the messages. However, I don't think storing this information in the snapshot could be useful. Since the snapshot is generated one time when the app is built, users will come later with probably new transactions as they use the bridge which won't be included in the snapshot.

Reply:
Yeah, OK. You are right -- adding this info to the snapshot has no much sense.